File: SortWare/FolderSelector.vb
Public Class FolderSelector
    Private settings As SortSettings
    Private returnDirectory As SortDirectory
    Public Sub New(ByVal _settings As SortSettings)

        ' This call is required by the designer.
        InitializeComponent()

        ' Add any initialization after the InitializeComponent() call.
        settings = _settings
        For Each f In settings.getList(SortSettings.dirType.PRESORTDIR)
            FoldersListBox.Items.Add(f)
        Next
    End Sub

    Public Function getSelectedDir() As SortDirectory
        Return returnDirectory
    End Function

    Private Sub DoneButton_Click(sender As Object, e As EventArgs) Handles doneButton.Click
        returnDirectory = DirectCast(FoldersListBox.SelectedItem, SortDirectory)
        Me.DialogResult = DialogResult.OK
        Me.Close()
    End Sub

    Private Sub CancelButton_Click(sender As Object, e As EventArgs) Handles cancelButton.Click
        Me.DialogResult = DialogResult.Cancel
        Me.Close()
    End Sub
End Class

File: SortWare/Interfaces/FolderSelector.cs
using System;
using System.Windows.Forms;

namespace SortWare
{
    public partial class FolderSelector
    {
        private SortSettings settings;
        private SortDirectory returnDirectory;
        public FolderSelector(SortSettings _settings)
        {

            // This call is required by the designer.
            InitializeComponent();

            // Add any initialization after the InitializeComponent() call.
            settings = _settings;
            foreach (var f in settings.getList(SortSettings.dirType.PRESORTDIR))
                FoldersListBox.Items.Add(f);
        }

        public SortDirectory getSelectedDir()
        {
            return returnDirectory;
        }

        private void DoneButton_Click(object sender, EventArgs e)
        {
            returnDirectory = (SortDirectory)FoldersListBox.SelectedItem;
            DialogResult = DialogResult.OK;
            Close();
        }

        private void CancelButton_Click(object sender, EventArgs e)
        {
            DialogResult = DialogResult.Cancel;
            Close();
        }

        private void FoldersListBox_SelectedIndexChanged(object sender, EventArgs e)
        {
            DoneButton_Click(sender, e);
        }
    }
}